ocviapy/__init__.py

```diff
@@ -595,58 +611,78 @@ def __init__(self, namespace, restype, name, watch_owned=False, watcher=None):
                 f"unable to check status of '{self.restype}' resources on this cluster"
             )
 
-    def _check_owned_resources(self, resource):
+    def raise_if_status_errors(self):
+        if self.status_errors:
+            combined_msg = "\n".join([f"* {msg}" for msg in self.status_errors])
+            log.error("Found resource status errors, details:\n%s", combined_msg)
+            raise StatusError("Found resource status errors, see logs for details")
+
+    def _owns_resource(self, resource):
         for owner_ref in resource.data["metadata"].get("ownerReferences", []):
             restype_matches = owner_ref["kind"].lower() == self.restype
             owner_uid_matches = owner_ref["uid"] == self.resource.uid
             if restype_matches and owner_uid_matches:
                 # this resource is owned by "self"
-                previously_observed = False
```

Comment:
This code block is unchanged, just moved into the new `_check_status_if_owned` func below for better readability